Public/Src/Engine/Cache/Artifacts/LocalDiskContentStore.cs

```diff
@@ -767,9 +771,13 @@ public bool TrackAbsentPath(AbsolutePath trackedParentPath, AbsolutePath absentC
             }
             else
             {
-                Possible<ContentHash, Failure> possiblyStored = await cache.TryStoreAsync(
-                    fileRealizationModes,
-                    expandedPath);
+                Possible<ContentHash, Failure> possiblyStored = await Helpers.RetryOnFailureAsync(
+                    async _ =>
+                    {
+                        return await cache.TryStoreAsync(

```

Comment:
>TryStoreAsync [](start = 43, length = 13)

Instead of doing it here, I think we should do it in `Public\Src\Engine\Cache.Plugin.CacheCore\CacheCoreArtifactContentCache.cs`.